File: notion_evidence_sync.js
// Notion Evidence Sync for Case 2024D007847
// Syncs current evidence analysis results to Notion database

const { Client } = require('@notionhq/client');
const fs = require('fs');
const path = require('path');
const crypto = require('crypto');

// Configuration - using existing vanguard onboarding config
const configPath = '/Users/nb/.claude/projects/-/legal/vangaurd/onboarding/notion_config.json';
const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));

// Initialize Notion client (will need 1Password integration for token)
const notion = new Client({
  auth: process.env.NOTION_TOKEN || 'ntn_...' // Will need actual token
});

// Generate ChittyID for evidence items
function generateChittyID(filename, index) {
  const version = '01';
  const geographic = '1';
  const location = 'CHI';
  const sequential = String(index + 2000).padStart(4, '0'); // Start at 2000 for case evidence
  const entityType = '3'; // Thing (evidence)
  const now = new Date();
  const yearMonth = now.getFullYear().toString().slice(-2) +
                   (now.getMonth() + 1).toString().padStart(2, '0');
  const category = 'L'; // Legal evidence

  // Calculate Mod-97 checksum
  const baseId = `${version}${geographic}${location}${sequential}${entityType}${yearMonth}${category}`;
  let remainder = 0;
  for (let char of baseId) {
    const val = /[A-Z]/.test(char) ? char.charCodeAt(0) - 55 : parseInt(char);
    remainder = (remainder * 10 + val) % 97;
  }
  const checksum = (98 - remainder).toString().padStart(2, '0');

  return `CT-${version}-${geographic}-${location}-${sequential}-${entityType}-${yearMonth}-${category}-${checksum}`;
}

// Calculate file hash
function calculateHash(filePath) {
  try {
    const fileBuffer = fs.readFileSync(filePath);
    const hashSum = crypto.createHash('sha256');
    hashSum.update(fileBuffer);
    return hashSum.digest('hex');
  } catch (error) {
    return 'file_not_accessible';
  }
}

// Current evidence analysis results to sync
function getCurrentEvidenceItems() {
  const evidenceItems = [];
  let index = 0;

  // Core analysis reports
  const reports = [
    {
      filename: 'FINAL_INTEGRATED_DISCOVERY_REPORT.md',
      classification: 'ANALYSIS_REPORT',
      description: 'Comprehensive integrated discovery analysis report'
    },
    {
      filename: 'evidence_analysis_complete.md',
      classification: 'ANALYSIS_SUMMARY',
      description: 'Evidence analysis completion summary'
    },
    {
      filename: 'openphone_critical_evidence.md',
      classification: 'COMMUNICATION_ANALYSIS',
      description: 'OpenPhone business communications analysis'
    },
    {
      filename: 'message_contradictions_summary.md',
      classification: 'CONTRADICTION_ANALYSIS',
      description: 'iMessage evidence contradictions report'
    },
    {
      filename: 'integrated_communication_timeline.md',
      classification: 'TIMELINE_ANALYSIS',
      description: 'Multi-platform communication timeline'
    },
    {
      filename: 'comprehensive_discovery_package.md',
      classification: 'DISCOVERY_PACKAGE',
      description: 'Complete discovery package documentation'
    },
    {
      filename: 'timeline_master.csv',
      classification: 'TIMELINE_DATA',
      description: 'Master timeline data in CSV format'
    },
    {
      filename: 'exhibit_index.csv',
      classification: 'EXHIBIT_INDEX',
      description: 'Complete exhibit indexing system'
    }
  ];

  reports.forEach(report => {
    const filePath = path.join('out', report.filename);
    evidenceItems.push({
      chitty_id: generateChittyID(report.filename, index++),
      filename: report.filename,
      file_path: filePath,
      file_hash: calculateHash(filePath),
      classification: report.classification,
      case_number: '2024D007847',
      legal_hold: true,
      verification_status: 'VERIFIED',
      description: report.description,
      analysis_date: new Date().toISOString(),
      evidence_type: 'ANALYSIS_OUTPUT'
    });
  });

  return evidenceItems;
}

// Sync evidence to Notion
async function syncEvidenceToNotion() {
  try {
    console.log('🔄 Starting Notion evidence sync...');

    const evidenceItems = getCurrentEvidenceItems();
    console.log(`📊 Found ${evidenceItems.length} evidence items to sync`);

    for (const item of evidenceItems) {
      try {
        // Create Notion page for each evidence item
        const response = await notion.pages.create({
          parent: { database_id: config.notion.database_id },
          properties: {
            'ChittyID': {
              title: [{ text: { content: item.chitty_id } }]
            },
            'File Name': {
              rich_text: [{ text: { content: item.filename } }]
            },
            'SHA-256 Hash': {
              rich_text: [{ text: { content: item.file_hash } }]
            },
            'Evidence Level': {
              select: { name: item.classification }
            },
            'Case Number': {
              rich_text: [{ text: { content: item.case_number } }]
            },
            'Legal Hold': {
              checkbox: item.legal_hold
            },
            'Status': {
              select: { name: item.verification_status }
            },
            'Description': {
              rich_text: [{ text: { content: item.description } }]
            },
            'Analysis Date': {
              date: { start: item.analysis_date }
            }
          }
        });

        console.log(`✅ Synced: ${item.filename} (${item.chitty_id})`);

        // Add delay to respect API limits
        await new Promise(resolve => setTimeout(resolve, 100));

      } catch (error) {
        console.error(`❌ Failed to sync ${item.filename}:`, error.message);
      }
    }

    console.log('🎯 Notion sync completed');

    // Generate sync report
    const syncReport = {
      timestamp: new Date().toISOString(),
      case_number: '2024D007847',
      items_synced: evidenceItems.length,
      sync_status: 'COMPLETED',
      database_id: config.notion.database_id,
      evidence_items: evidenceItems.map(item => ({
        chitty_id: item.chitty_id,
        filename: item.filename,
        classification: item.classification
      }))
    };

    // Save sync report
    fs.writeFileSync('out/notion_sync_report.json', JSON.stringify(syncReport, null, 2));
    console.log('📄 Sync report saved to out/notion_sync_report.json');

  } catch (error) {
    console.error('💥 Notion sync failed:', error.message);

    // Check if it's authentication issue
    if (error.code === 'unauthorized') {
      console.log('🔑 Authentication required - ensure NOTION_TOKEN environment variable is set');
      console.log('💡 Use: export NOTION_TOKEN="your_notion_integration_token"');
    }
  }
}

// Test mode - just show what would be synced
function testSync() {
  console.log('🧪 Test Mode - Evidence items that would be synced:');
  const evidenceItems = getCurrentEvidenceItems();

  evidenceItems.forEach((item, index) => {
    console.log(`${index + 1}. ${item.chitty_id} - ${item.filename}`);
    console.log(`   Classification: ${item.classification}`);
    console.log(`   Hash: ${item.file_hash.substring(0, 16)}...`);
    console.log('');
  });

  console.log(`📊 Total items ready for sync: ${evidenceItems.length}`);
}

// Main execution
if (require.main === module) {
  const args = process.argv.slice(2);

  if (args.includes('--test')) {
    testSync();
  } else if (args.includes('--sync')) {
    syncEvidenceToNotion();
  } else {
    console.log('Notion Evidence Sync for Case 2024D007847');
    console.log('');
    console.log('Usage:');
    console.log('  node notion_evidence_sync.js --test   # Show what would be synced');
    console.log('  node notion_evidence_sync.js --sync   # Perform actual sync');
    console.log('');
    console.log('Environment variables needed:');
    console.log('  NOTION_TOKEN - Your Notion integration token');
    console.log('');
    console.log('📊 Current evidence analysis ready for sync');
  }
}

module.exports = {
  syncEvidenceToNotion,
  getCurrentEvidenceItems,
  generateChittyID
};

File: notion_evidence_sync.py
#!/usr/bin/env python3

"""
Notion Evidence Sync for Arias v. Bianchi
Syncs evidence analysis to Notion for case management
"""

import os
import json
import pandas as pd
from datetime import datetime
from pathlib import Path
import click

# Try to import notion client
try:
    from notion_client import Client
    NOTION_AVAILABLE = True
except ImportError:
    NOTION_AVAILABLE = False

@click.command()
@click.option('--notion-token', envvar='NOTION_TOKEN', help='Notion integration token')
@click.option('--database-id', envvar='NOTION_DATABASE_ID', help='Notion database ID for evidence')
@click.option('--test', is_flag=True, help='Test mode - show what would be synced')
@click.option('--case-name', default='Arias v. Bianchi', help='Case name')
@click.option('--case-id', default='2024D007847', help='Case ID')
def sync_to_notion(notion_token, database_id, test, case_name, case_id):
    """Sync evidence data to Notion"""

    if not NOTION_AVAILABLE:
        click.echo("❌ Notion client not installed")
        click.echo("Run: pip3 install notion-client")
        return

    if not notion_token:
        click.echo("❌ No Notion token provided!")
        click.echo("1. Go to https://www.notion.so/my-integrations")
        click.echo("2. Create new integration")
        click.echo("3. Copy the token")
        click.echo("4. Set NOTION_TOKEN environment variable")
        return

    if not database_id:
        click.echo("❌ No database ID provided!")
        click.echo("1. Create a database in Notion")
        click.echo("2. Share it with your integration")
        click.echo("3. Copy the database ID from the URL")
        click.echo("4. Set NOTION_DATABASE_ID environment variable")
        return

    click.echo(f"📊 Syncing evidence for {case_name} to Notion")

    # Load evidence data
    timeline_path = Path("out/timeline_master.csv")
    exhibit_path = Path("out/exhibit_index.csv")

    if not timeline_path.exists():
        click.echo(f"❌ Timeline file not found: {timeline_path}")
        return

    timeline_df = pd.read_csv(timeline_path)
    exhibit_df = pd.read_csv(exhibit_path) if exhibit_path.exists() else pd.DataFrame()

    click.echo(f"📥 Loaded {len(timeline_df)} timeline items")
    click.echo(f"📥 Loaded {len(exhibit_df)} exhibit items")

    if test:
        click.echo("\n🧪 TEST MODE - Preview of data to sync:")

        # Show summary statistics
        click.echo(f"\n📊 Evidence Summary:")
        click.echo(f"   • Total items: {len(timeline_df)}")
        click.echo(f"   • Exhibits: {len(exhibit_df)}")

        # Show event types
        if 'type' in timeline_df.columns:
            type_counts = timeline_df['type'].value_counts()
            click.echo(f"\n📋 Event Types:")
            for event_type, count in type_counts.items():
                click.echo(f"   • {event_type}: {count}")

        # Show recent items
        click.echo(f"\n📅 Recent Evidence Items:")
        recent_items = timeline_df.head(5)
        for _, item in recent_items.iterrows():
            date = item.get('timestamp', item.get('date', 'Unknown'))
            desc = str(item.get('description', item.get('content', 'No description')))[:100]
            click.echo(f"   • {date}: {desc}")

        # Show exhibit samples
        if not exhibit_df.empty:
            click.echo(f"\n🏷️  Sample Exhibits:")
            for _, exhibit in exhibit_df.head(3).iterrows():
                exhibit_id = exhibit.get('exhibit_id', 'Unknown')
                desc = str(exhibit.get('description', 'No description'))[:80]
                click.echo(f"   • {exhibit_id}: {desc}")

        click.echo(f"\n✅ Test complete. Use --test=false to perform actual sync")
        return

    # Initialize Notion client
    try:
        notion = Client(auth=notion_token)
        click.echo("✅ Connected to Notion")
    except Exception as e:
        click.echo(f"❌ Failed to connect to Notion: {e}")
        return

    # Test database access
    try:
        database = notion.databases.retrieve(database_id=database_id)
        click.echo(f"✅ Found database: {database['title'][0]['plain_text']}")
    except Exception as e:
        click.echo(f"❌ Cannot access database: {e}")
        click.echo("Make sure the database is shared with your integration")
        return

    # Sync evidence items
    click.echo("\n📤 Syncing evidence to Notion...")

    synced_count = 0
    error_count = 0

    # Sync timeline items (limit to recent items to avoid rate limits)
    recent_timeline = timeline_df.head(50)  # Sync most recent 50 items

    for _, item in recent_timeline.iterrows():
        try:
            # Prepare properties for Notion
            properties = {
                "Title": {
                    "title": [
                        {
                            "text": {
                                "content": str(item.get('description', 'Evidence Item'))[:100]
                            }
                        }
                    ]
                },
                "Case": {
                    "rich_text": [
                        {
                            "text": {
                                "content": case_name
                            }
                        }
                    ]
                },
                "Type": {
                    "select": {
                        "name": str(item.get('type', 'other'))
                    }
                },
                "Date": {
                    "date": {
                        "start": str(item.get('timestamp', item.get('date', datetime.now().isoformat())))[:10]
                    }
                },
                "Source": {
                    "rich_text": [
                        {
                            "text": {
                                "content": str(item.get('source', item.get('source_file', 'Unknown')))[:100]
                            }
                        }
                    ]
                },
                "Confidence": {
                    "select": {
                        "name": str(item.get('confidence', 'medium'))
                    }
                }
            }

            # Add content if available
            content_text = str(item.get('content', item.get('description', '')))
            if content_text and content_text != 'nan':
                properties["Content"] = {
                    "rich_text": [
                        {
                            "text": {
                                "content": content_text[:2000]  # Notion limit
                            }
                        }
                    ]
                }

            # Create page in Notion
            notion.pages.create(
                parent={"database_id": database_id},
                properties=properties
            )

            synced_count += 1

        except Exception as e:
            error_count += 1
            if error_count <= 3:  # Show first few errors
                click.echo(f"⚠️  Error syncing item: {e}")

    click.echo(f"\n✅ Sync complete!")
    click.echo(f"   • Synced: {synced_count} items")
    click.echo(f"   • Errors: {error_count} items")

    # Generate sync report
    report = {
        "sync_timestamp": datetime.now().isoformat(),
        "case_name": case_name,
        "case_id": case_id,
        "total_timeline_items": len(timeline_df),
        "total_exhibits": len(exhibit_df),
        "synced_items": synced_count,
        "sync_errors": error_count,
        "notion_database_id": database_id
    }

    # Save sync report
    with open("out/notion_sync_report.json", "w") as f:
        json.dump(report, f, indent=2)

    click.echo(f"📄 Sync report saved to: out/notion_sync_report.json")

    # Show next steps
    click.echo(f"\n🔗 View your evidence in Notion:")
    click.echo(f"   https://notion.so/{database_id.replace('-', '')}")

if __name__ == '__main__':
    sync_to_notion()